puppet/modules/koji/manifests/mirror.pp

```diff
@@ -0,0 +1,48 @@
+# Basic mirror for Koji
+class koji::mirror {
+  ensure_packages(['dnf-plugins-core'])
+
+  file { '/srv/mirror':
+    ensure => directory,
+    owner  => 'root',
+    group  => 'root',
+    mode   => '0755',
+  }
+
+  if $facts['os']['selinux']['enabled'] {
+    include selinux
+
+    selinux::fcontext { '/srv/mirror':
+      seltype  => 'httpd_sys_content_t',
+      pathspec => '/srv/mirror(/.*)?',
+      notify   => File['/srv/mirror'],
+    }
+  }
+
+  file { '/etc/reposync.conf':
+    content => file('koji/mirror_reposync.conf'),
+    mode    => '0644',
+  }
+
+  file { '/etc/cron.weekly/reposync':
+    content => file('koji/mirror_reposync.cron'),
+    mode    => '0755',
+    require => [File['/srv/mirror', '/etc/reposync.conf'], Package['dnf-plugins-core']],
+  }
+
+  class { 'apache':
+    default_vhost => false,
+  }
+
+  apache::vhost { 'mirror.koji.theforeman.org':
+    serveraliases => ['mirror.koji.aws.theforeman.org'],
+    port          => 80,
+    docroot       => '/srv/mirror',
+    directories   => [
+      {
+        'path'    => '/srv/mirror',
+        'require' => 'ip 127.0.0.1',
```

Comment:
Might be nice to have the name, aliases and IP requirement as class parameters.

Reply:
yeah, this part isn't done yet, I need to figure out the AWS VPC layout for that